From 38857cfd34ba919e8c711be2584ea06997e2f692 Mon Sep 17 00:00:00 2001
From: wuzhouhui <wuzhouhui250@gmail.com>
Date: Mon, 10 Apr 2017 20:43:37 +0800
Subject: [PATCH 41/95] Subcmd clean: remove files only if option --force
 enabled

Removing unversioned files is a dangerous operation, we should
think twice before do it.
---
 subversion/svn/svn.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/subversion/svn/svn.c b/subversion/svn/svn.c
index c70156e..c08cc66 100644
--- a/subversion/svn/svn.c
+++ b/subversion/svn/svn.c
@@ -630,7 +630,7 @@ const svn_opt_subcommand_desc2_t svn_cl__cmd_table[] =
 
   { "clean", svn_cl__clean, {0}, N_
     ("Remove unversioned files and directories.\n"
-     "usage: clean [PATH...]\n"), { 'q', opt_no_ignore } },
+     "usage: clean [PATH...]\n"), { 'q', opt_no_ignore, opt_force } },
 
   { "cleanup", svn_cl__cleanup, {0}, N_
     ("Either recover from an interrupted operation that left the working copy locked,\n"
@@ -2895,6 +2895,16 @@ sub_main(int *exit_code, int argc, const char *argv[], apr_pool_t *pool)
                                 "are mutually exclusive"));
     }
 
+  /* Think twice before removing unversioned files. */
+  if (opt_state.force == FALSE && subcommand->cmd_func == svn_cl__clean)
+  {
+    svn_error_clear(
+        svn_cmdline_fprintf(stderr, pool,
+          _("Add option --force if you really want to remove files.\n")));
+    *exit_code = EXIT_FAILURE;
+    return SVN_NO_ERROR;
+  }
+
   /* --trust-* options can only be used with --non-interactive */
   if (!opt_state.non_interactive)
     {
-- 
1.8.3.1

